src/core/localize.rkt

```diff
@@ -191,73 +191,174 @@
+
+  ;; Combine loops over pcontext to use for/vectors?
+  (define exacts-from-points
+    (for/vector #:length (pcontext-length (*pcontext*))
+                ([(pt ex) (in-pcontext (*pcontext*))])
+      (list->vector (apply subexprs-fn pt))))
+  (define actuals-from-points
+    (for/vector #:length (pcontext-length (*pcontext*))
+                ([(pt ex) (in-pcontext (*pcontext*))])
+      (apply actual-value-fn pt)))
+
+  (define (compute-abs-error spec exact ctx pt repr)
+    ;; TODO compute in batches and evalutate propigated errors from rival.
+    (define true-error (first (apply (eval-progs-real (list `(- ,spec ,exact)) (list ctx)) pt)))
+    (define bf-true-error ((representation-repr->bf repr) true-error))
+    (define abs-error (bfabs bf-true-error))
+    (define abs-error-out (value->json (bigfloat->flonum abs-error) repr))
+    abs-error-out)
+
+  (define (absolute-error-for i exact pt repr)
+    (define approx-spec (vector-ref spec-vec i))
+    (define approx-ctx (vector-ref ctx-vec i))
+    (compute-abs-error approx-spec exact approx-ctx pt repr))
+
+  (define previous_node_if? #f)
+
+  (define data-hash (make-hash))
+
+  (for ([(pt ex) (in-pcontext (*pcontext*))]
+        [exacts (in-vector exacts-from-points)]
+        [actuals (in-vector actuals-from-points)])
+    (for ([expr-syntax (in-list (all-subexpressions fpcore))]
+          [expr (in-list exprs-list)]
+          [current-spec (in-vector spec-vec)]
+          [current-ctx (in-list ctx-list)]
+          [root (in-vector roots)]
+          [actual (in-vector actuals)]
+          [exact (in-vector exacts)])
+      (define node (vector-ref nodes root))
+      (define current-repr (hash-ref repr-hash expr))
+      (define local-error
+        (match node
+          [(? literal?) 1]
+          [(? variable?) 1]
+          [(approx _ impl)
+           (define repr (repr-of expr ctx))
+           (ulp-difference exact (vector-ref exacts (vector-member impl roots)) repr)]
+          [`(if ,c ,ift ,iff) 1]
+          [(list f args ...)
+           (define repr (impl-info f 'otype))
+           (define argapprox
+             (for/list ([idx (in-list args)])
+               (vector-ref exacts (vector-member idx roots))))
+           (define approx (apply (impl-info f 'fl) argapprox))
+           (ulp-difference exact approx repr)]))
+      (define abs-error
+        (match node
+          [(? literal?) (compute-abs-error current-spec exact current-ctx pt current-repr)]
+          [(? variable?) 0]
+          [(approx _ impl) (absolute-error-for impl exact pt current-repr)]
+          [`(if ,c ,ift ,iff)
+           (set! previous_node_if? #t)
+           (if exact
+               (absolute-error-for ift exact pt current-repr)
+               (absolute-error-for iff exact pt current-repr))]
+          [(list f args ...)
+           (define local previous_node_if?)
+           (when previous_node_if?
+             (set! previous_node_if? #f))
+           (if local
```

Comment:
```suggestion
           (if (equal? (representation-type (impl-info f 'otype)) 'bool)
```

I think this is what you want, then you can dump all the `previous_node_if?` stuff.